swap/src/bin/asb.rs

```diff
@@ -272,27 +281,39 @@ async fn main() -> Result<()> {
     Ok(())
 }
 
-async fn init_bitcoin_wallet(
+async fn init_bitcoin_wallets(
     config: &Config,
     seed: &Seed,
     env_config: swap::env::Config,
-) -> Result<bitcoin::Wallet> {
-    debug!("Opening Bitcoin wallet");
-    let wallet_dir = config.data.dir.join("wallet");
-
-    let wallet = bitcoin::Wallet::new(
+) -> Result<(bitcoin::Wallet, bitcoin::Wallet)> {
+    debug!("Opening Bitcoin wallets");
+    let wallet_dir = config.data.dir.join("internal_wallet");
+    let internal_wallet = bitcoin::Wallet::new(
         config.bitcoin.electrum_rpc_url.clone(),
         &wallet_dir,
         seed.derive_extended_private_key(env_config.bitcoin_network)?,
         env_config,
         config.bitcoin.target_block,
     )
-    .await
-    .context("Failed to initialize Bitcoin wallet")?;
+    .context("Failed to initialize protocol Bitcoin wallet")?;
+
+    let xpub = match config.bitcoin.xpub {
+        None => seed.derive_extended_public_key(env_config.bitcoin_network)?,
+        Some(key) => key,
+    };
+    let wallet_dir = config.data.dir.join("personal_wallet");
+    let personal_wallet = bitcoin::wallet::Wallet::new_readonly(
+        config.bitcoin.electrum_rpc_url.clone(),
+        &wallet_dir,
+        xpub,
+        env_config,
+    )
+    .context("Failed to initialize personal Bitcoin wallet")?;
 
-    wallet.sync().await?;
+    internal_wallet.sync().await?;
+    personal_wallet.sync().await?;
```

Comment:
The user will have to tell us the re-scan height of the key, otherwise we would assume the key hasn't been used before and we will re-use addresses.

Reply:
I disagree with your conclusion: 

If a user provides an xpub key he really should not use this xpub otherwise to get addresses. 
Otherwise we cannot guarantee that we won't re-use addresses. Even if we ask the user for a height. 

Also, this function does not care about the height. 
Does BDK support the feature of only scanning from a particular height?

Reply:
> If a user provides an xpub key he really should not use this xpub otherwise to get addresses.

Why not? It seems perfectly reasonable to wanting to pool together funds into a single wallet from different sources?

> Does BDK support the feature of only scanning from a particular height?

I thought it did but apparently it doesn't. https://docs.rs/bdk/0.7.0/bdk/wallet/struct.Wallet.html#method.sync

The feature of rescan height is actually coupled to the individual blockchain backend in BDK. Since we are using electrum, it doesn't matter.

Reply:
> Why not? It seems perfectly reasonable to wanting to pool together funds into a single wallet from different sources?

It's mostly a privacy concern: Anyone holding your xpub can derive addresses and hence can learn the balances. Meaning, the Asb or Cli can both learn about all your funds and transaction history unrelated to our applications. 

There is also the concern about security of address re-use which we cannot guarantee if addresses are being created outside of our control.

Reply:
Fair. As we don't support any other backends than Electrum for now, it is not a performance concern so we don't need the rescan height actually.